Nations update

DeStup opened the pull request:
I compared Rekmod with the PC version lekmod 33.7 and found that many units have promotions instead uniqueness, that keeps after upgrade. 
Some uniques reimplemented.
Changes by nation:
- England
- Goths
- Ayybids
- Armenia
- Assyria
- Akkad
- Boers
- Brazil
- Gaul
- Finland
- Japan
- Jerusalem
- Carthagen
There is still a list that has not been included here yet, it will be in the next PR

also several fixes for 
- Granary, 
- Khambar, 
- Kampong, 
- Mead Hall, 
- Samurai, 
- Ocupada Estable

SeventhM commented:
> - Granary (and replacements)
> - Stable (and replacements)

I'm making 3 requests
1. I know the question has been asked on the Lekmod server before, but I would like to know their reasoning on this succinctly and if they feel that is the better choice or if they feel it's more or less a minor thing to discuss. Otherwise, I'm more than willing to pretend it is an intentional deviation. I do realize it has something to do with a feeling that granaries do too much, but it seems like an awkward game decision choice that I'm actually more willing to stand my ground here and directly challenge them to confirm it's a better setup. (Fun fact, there's some newer promotions with stuff like the anti naval ranged promotion line that I also took creative liberties on as it looks like just an oversight by the Lekmod mod team. I haven't checked if they fixed them, but I don't remember seeing it in any change log)
2. I would like to be pinged on discord to their answer to such a question, not here
3. I would like that split to a different PR to make merging this easier
Also, you didn't fix the other stable replacements (notably the base stable itself)

Give me some time to double check the promotions. Need to make sure they are/aren't permanent

DeStup commented:
1. The funny thing is that in PR I mixed up the granary and changed the Tongo Mala'e instead of the Granary.
2.As for the change, I agree that it looks strange. I accidentally came across this difference when I was studying the peculiarities of nations. I rolled back so as not to block the other more important changes.
3. I checked the stables, added a change

DeStup commented:
Updated 
- Ngao Mbeba
- Manchuria uniq
- Maurya description
- Pillar of Ashoka maintenance
- Double horse for Mongolia after Chivalry
- Khan movement 4->5
- Berber rider
- Ski infantry
- Nubia uniq
- Sipahi uniq
- Phoenicia uniq fix
- Square Sail Ship new promotion
- fix portugal uniq (discussed)
- new promotion for Landwehr, deleted old uniq
- new uniq for Sweden,
- new promotion for Carolean
- romania new uniq
- Vanator new promotion
- Comanche Raider new promotion
- Phalanx new promotion
- Renamed Enhanced Flank Attack
- Serai buff
- Kuva-yi Milliye new promotion and new uniq
- Knyaz Court balanced for Quick game
- new uniq for Vietnam
- new uniq for Viet Kong

DeStup commented:
i used to check this document (32.8)
https://docs.google.com/document/d/e/2PACX-1vSUkKuhdURg8zS2P4nMHRjXvkp-Tg5QPxvMfaXVoIiyfBgPQxqVaxAeJG9JfZN6HAq9SjYE4NLxjLyk/pub

And i checked every change in Civilization V with 33.7 lekmod version.
I may have made a mistake somewhere, but the list turned out to be impressive.

SeventhM commented on jsons/Buildings.json:
```diff
@@ -1113,6 +1117,7 @@
         "name": "Mead Hall",
         "replaces": "Colosseum",
         "uniqueTo": "Franks",
+        "cost": 50,
```
All stats should be correct for standard game speed, not quick. Actual cost was already correct (75), this change is incorrect according to Lekmod xml

DeStup replied:
then we have the wrong price for the colosseum... Because Mead Hall should be 25% cheaper.

DeStup replied:
забавно, я не заметил этого в первый раз...
![image](https://github.com/user-attachments/assets/029b2f00-67f4-477c-90df-eb6b22597a4a)
![image](https://github.com/user-attachments/assets/406cb822-c267-470d-8393-926e7d3c62b4)

DeStup replied:
32f2c7ceca283f8cfe16a9f2e2ede495ebd8abd2

SeventhM commented on jsons/Buildings.json:
```diff
@@ -1488,7 +1493,7 @@
         "percentStatBonus": {"gold": 25},
         "uniques": [
-            "Gain [20] [Gold] <(modified by game speed)> <upon gaining a [Great Person] unit>", // should be on expending... give me some time to work on it
+            "Gain [30] [Gold] <(modified by game speed)> <upon gaining a [Great Person] unit>", // should be on expending... give me some time to work on it
```
Original benefit was correct according to the xml. If you think the benefit was wrong compared to actual Lekmod, then it means it isn't modified by game speed. I need to do tests on that. But it is in fact 20, not 30

SeventhM replied:
Can confirm, it is not modified by game speed. Apparently, Mausoleum of Halicarnassus isn't either

DeStup replied:
32f2c7ceca283f8cfe16a9f2e2ede495ebd8abd2

SeventhM commented on jsons/Nations.json:
```diff
@@ -2142,7 +2148,7 @@
         "innerColor": [0,0,0],
         "uniqueName": "Ta-Seti",
-        "uniques": ["Free [Apedemaks Bow] appears <upon discovering [Pottery] technology>"],
+        "uniques": ["Free [Apedemaks Bow] appears <upon discovering [Agriculture] technology>"],
```
If I remember correctly, this doesn't function. Remove the the conditional and then it should work

DeStup replied:
it worked, but okay, I'll just delete the condition.

SeventhM replied:
Really? That's surprising tbh

SeventhM commented on jsons/Nations.json:
```diff
@@ -2299,7 +2305,7 @@
         "uniqueName": "Skillful Traders",
         "uniques": [
-            "Gain [50] [Gold] <(modified by game speed)> <upon founding a city> <after discovering [Optics]>",
+            "Gain [75] [Gold] <(modified by game speed)> <upon founding a city> <after discovering [Optics]>",
```
Looks like a Lekmod bug. In Lekmod, it is in fact 50 Gold, but it is \*not\* modified by game, which looks incorrect. Will accept this change, but keep this in mind

SeventhM commented on jsons/Units.json:
```diff
@@ -691,11 +693,10 @@
             "[+200]% Strength <vs cities> <when attacking>",
-            "No defensive terrain bonus",
-            "[-1] Sight"
+            "No defensive terrain bonus"
```
Side note: This unit should only be able to attack cities. Also missing mechanic that gives an attack bonus when next to a city. Otherwise, changes in the PR looks correct

DeStup replied:
added "Can only attack [City] units"

SeventhM commented:
Half surprised the promotions look to actually be promotions and not just uniques. So, yeah, main thing is fixing those stuff, and extracting stable/granary changes to a different branch and doing that as a different PR

SeventhM commented:
And by all of the granary/stable changes, I do mean all of them. including the Ducal Stable/Ocupada Estable/Kambar ones (those I also manually changed to match the seeming intent)

DeStup commented:
> And by all of the granary/stable changes, I do mean all of them. including the Ducal Stable/Ocupada Estable/Kambar ones (those I also manually changed to match the seeming intent)

Okay, I'll do a separate PR on the stables and granary later.
It seems to have rolled back everything here

SeventhM commented on jsons/Buildings.json:
```diff
@@ -988,6 +988,7 @@
         "replaces": "Caravansary",
         "cost": 100,
         "hurryCostModifier": 25,
+        "gold": 2,
```
Serai does not inherently make gold as far as I'm aware

DeStup replied:
ye, my bad 
fixed 6c569a493488549c4e4742d5e03c02935a91d809

SeventhM commented on jsons/Units.json:
```diff
@@ -3094,7 +3088,7 @@
             "Uncapturable"
         ],
-        "movement": 4
+        "movement": 5
```
Hmm... Apparently it's 7

DeStup replied:
> Hmm... Apparently it's 7

https://github.com/ravignir/RekMOD/pull/251/commits/447f32e67a4b669b253a73338d66346cdee304de

SeventhM commented:
From what I can tell, everything looks correct. I need to double check colosseums to see if they're an exception or if more buildings needs there cost increased, but I'm not going to hold up this PR for that. Merging
